build/initialize-pipeline copy.ps1

```diff
@@ -0,0 +1,19 @@
+# Get major, minor and patchVersions
+[xml]$XMLContents = [xml](Get-Content -Path ".\common.props")
+$XMLContents.GetElementsByTagName("MajorVersion") |  ForEach-Object {
+  $majorVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=MajorVersion;isOutput=true]$majorVersion"
+  Write-Host "Setting 'MajorVersion' to $majorVersion"
+}
+
+$XMLContents.GetElementsByTagName("MinorVersion") |  ForEach-Object {
+  $minorVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=MinorVersion;isOutput=true]$minorVersion"
+  Write-Host "Setting 'MinorVersion' to $minorVersion"
+}
+
+$XMLContents.GetElementsByTagName("PatchVersion") |  ForEach-Object {
+  $patchVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=PatchVersion;isOutput=true]$patchVersion"
+  Write-Host "Setting 'PatchVersion' to $patchVersion"
+}
```

Comment:
I'd like to take this change in v4 but I have more complex version logic there b/c of the "preview" suffix I need to make.

To handle this, I made a target explicitly for creating a version.txt file that has the calculated version from common.props: https://github.com/Azure/azure-functions-host/blob/v4.x/build/common.props#L25-L31. Then my build reads it.

It's similar to your approach (read common.props for the "truth")... but allows the suffix and treats the common.props even more as the "source of truth". 

I wonder if we can merge these approaches... 

That being said, I'm completely okay with this right now, then we can follow up with something that may add some flexibility for v4 so we can share the logic going forward.

Reply:
Opened follow up issue here https://github.com/Azure/azure-functions-host/issues/7376 .

Reply:
Thanks!